nixos/modules/services/torrent/deluge.nix

```diff
@@ -5,8 +5,32 @@ with lib;
 let
   cfg = config.services.deluge;
   cfg_web = config.services.deluge.web;
+
   openFilesLimit = 4096;
 
+  listToRange = (x: { from = elemAt x 0; to = elemAt x 1; } );
+
+  configDir = "${cfg.dataDir}/.config/deluge";
+  configFile = pkgs.writeText "core.conf" (builtins.toJSON cfg.config);
+  declarativeLockFile = "${configDir}/.declarative";
+
+  preStart = (if cfg.declarative then ''
```

Comment:
Something minor here too: The parentheses around the whole expression can be dropped.